sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/EquivalentExpressions.scala

```diff
@@ -72,7 +72,10 @@ class EquivalentExpressions {
     val skip = expr.isInstanceOf[LeafExpression] ||
       // `LambdaVariable` is usually used as a loop variable, which can't be evaluated ahead of the
       // loop. So we can't evaluate sub-expressions containing `LambdaVariable` at the beginning.
-      expr.find(_.isInstanceOf[LambdaVariable]).isDefined
+      expr.find(_.isInstanceOf[LambdaVariable]).isDefined ||
+      // `PlanExpression` wraps query plan. To compare query plans of `PlanExpression` on executor,
+      // can cause unexpected error.
```

Comment:
If there isn't any other reason, shall we mention NPE specifically instead of `unexpected error`? Both SPARK-29239 and SPARK-29221 are due to NPE.

Reply:
Ok, I updated this.

Reply:
How about adding ` including NPE` after `unexpected error`? IMHO `unexpected error` is actually correct (we can't predict which error we will get), but it would help much if we enumerate known errors as well.

Reply:
Oh I realized my browser didn't show the comment from @viirya . It's just a 2 cents.